data-prepper-plugins/cloudwatch-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/CloudwatchMetricsSource.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source;
+
+import org.opensearch.dataprepper.aws.api.AwsCredentialsSupplier;
+import org.opensearch.dataprepper.buffer.common.BufferAccumulator;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+
+import java.time.Duration;
+import java.util.Collection;
+
+import org.opensearch.dataprepper.model.source.coordinator.SourceCoordinator;
+import org.opensearch.dataprepper.model.source.coordinator.UsesSourceCoordination;
+import software.amazon.awssdk.services.cloudwatch.model.Dimension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchClient;
+import java.util.ArrayList;
+import java.util.Objects;
+
+
+/**
+ *  An implementation of cloudwatch metrics source class to Scrape the metrics using GetMetricData API
+ */
+@DataPrepperPlugin(name = "cloudwatch", pluginType = Source.class, pluginConfigurationType = CloudwatchMetricsSourceConfig.class)
+public class CloudwatchMetricsSource implements Source<Record<Event>>, UsesSourceCoordination {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CloudwatchMetricsSource.class);
+
+    private final Collection<Dimension> dimensionCollection;
```

Comment:
I don't think this needs to be a field. It should be a local variable where you use it. Otherwise, this is not thread-safe and may not work with all metrics.

Reply:
this has been fixed.